ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/TimeoutHandler.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.handlers;
+
+import static java.util.Collections.emptyMap;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.context.ContextKey;
+
+import java.util.Map;
+import java.util.Optional;
+
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+public class TimeoutHandler {
+
+  public static Handler<RoutingContext> handler(
+      final Optional<TimeoutOptions> globalOptions, final boolean decodeJSON) {
+    return handler(globalOptions, emptyMap(), decodeJSON);
+  }
+
+  public static Handler<RoutingContext> handler(
+      final Optional<TimeoutOptions> globalOptions,
+      final Map<String, TimeoutOptions> timeoutOptionsByMethod,
+      final boolean decodeJSON) {
+    assert timeoutOptionsByMethod != null;
+    return ctx -> {
+      try {
```

Comment:
Should this be moved to a method to make the lambda simpler?

Reply:
Makes sense.

Reply:
Done.